tests/modules/apigee_mock_client/src/MockHandlerStack.php

```diff
@@ -0,0 +1,164 @@
+      $this->responses = Yaml::parseFile($module_path . '/response_catalog.yml')['responses'];
+    }
+    // Loop through responses and add each one.
+    foreach ((array) $response_ids as $index => $item) {
+      // The catalog id should either be the item itself or the keys if an associative array has been passed.
+      $id = !is_array($item) ? $item : $index;
+      // Body text can have elements replaced in it for certain values. See: http://php.net/strtr
+      $context = is_array($item) ? $item : [];
+      $context['org_name'] = isset($context['org_name']) ? $context['org_name'] : $org_name;
+
+      // Add the default headers if headers aren't defined in the response catalog.
+      $headers = isset($this->responses[$id]['headers']) ? $this->responses[$id]['headers'] : [
+        'content-type' => 'application/json;charset=utf-8'
+      ];
+      // Set the default status code.
+      $status_code = !empty($this->responses[$id]['status_code']) ? $this->responses[$id]['status_code'] : 200;
+      $status_code = !empty($context['status_code']) ? $context['status_code'] : $status_code;
+
+      // Render the response content.
+      $template = str_replace('_', '-', $id) . '.json.twig';
+      $content = $this->twig->getLoader()->exists($template) ? $this->twig->render($template, $context) : '';
+
+      // Make replacements inside the response body and append the response.
+      $this->append(new Response(
+        $status_code,
+        $headers,
+        $content
+      ));
+    }
+
+    return $this;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __invoke(RequestInterface $request, array $options) {
+    try {
+      // Grab an item from the database queue and append it to the in-memory queue.
+      if ($item = $this->database_queue->claimItem()) {
+        parent::append(new Response(
+          $item->data['status'],
+          $item->data['headers'],
+          $item->data['body']
+        ));
+        $this->database_queue->deleteItem($item);
+      }
+
+      return parent::__invoke($request, $options);
+    } catch (\Exception $ex) {
+      // Fake a client error so the error gets reported during testing. The
+      // `apigee_edge_user_presave` is catching errors in a way that this error
+      // never gets reported to PHPUnit. This won't work for all use cases.
+      //
+      // @todo: Find another way to ensure an exception gets registered with PHPUnit.
+      throw new ClientErrorException(new Response(500, [
+        'Content-Type' => 'application/json',
+      ], '{"code": "4242", "message": "'.$ex->getMessage().'"}'), $request, $ex->getMessage(), $ex->getCode(), $ex);
+    }
+  }
+
+  public function append() {
+    foreach (func_get_args() as $value) {
+      if ($value instanceof ResponseInterface) {
+
+        // Append items to a database queue so they can be accessed in process isolated environments (functional tests).
```

Comment:
> Lines containing comments (including docblocks) must wrap as close to 80 characters as possible without going over, with a few exceptions (noted in the Tag Reference below).
https://www.drupal.org/docs/develop/standards/api-documentation-and-comment-standards